chains/evm/contracts/ocr/SignatureQuorumVerifier.sol

```diff
@@ -2,150 +2,169 @@
 pragma solidity ^0.8.4;
 
 import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
-import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
 
 import {EnumerableSet} from
   "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v5.0.2/contracts/utils/structs/EnumerableSet.sol";
 
-/// @notice Onchain verification of reports from the offchain reporting protocol with multiple OCR plugin support.
-contract SignatureQuorumVerifier is ITypeAndVersion, Ownable2StepMsgSender {
+contract SignatureQuorumVerifier is Ownable2StepMsgSender {
   using EnumerableSet for EnumerableSet.AddressSet;
+  using EnumerableSet for EnumerableSet.Bytes32Set;
 
-  string public constant override typeAndVersion = "SignatureQuorumVerifier 1.7.0-dev";
-
-  /// @notice Triggers a new run of the offchain reporting protocol.
   /// @param configDigest configDigest of this configuration.
   /// @param signers ith element is address ith oracle uses to sign a report.
-  /// @param F maximum number of faulty/dishonest oracles the protocol can tolerate while still working correctly.
-  event ConfigSet(bytes32 configDigest, address[] signers, uint8 F);
+  /// @param F maximum number of faulty/dishonest singers the protocol can tolerate while still working correctly.
+  event ConfigSet(bytes32 indexed configDigest, address[] signers, uint8 F);
+  event ConfigRevoked(bytes32 indexed configDigest);
 
   error InvalidConfig();
-  error ConfigDigestMismatch(bytes32 expected, bytes32 actual);
+  error InvalidConfigDigest(bytes32 configDigest);
+  error ConfigDigestAlreadyExists(bytes32 configDigest);
   error ForkedChain(uint256 expected, uint256 actual);
   error WrongNumberOfSignatures();
   error SignaturesOutOfRegistration();
   error UnauthorizedSigner();
   error NonUniqueSignatures();
   error OracleCannotBeZeroAddress();
 
-  struct SignatureConfigConfig {
-    bytes32 configDigest;
-    uint8 F; //  maximum number of faulty/dishonest oracles the system can tolerate.
-    uint8 n; //  number of configured signers.
-  }
-
   /// @notice Args to update an OCR Config.
   struct SignatureConfigArgs {
     bytes32 configDigest; // The new config digest.
     uint8 F; // Maximum number of faulty/dishonest oracles.
     address[] signers; // signing address of each oracle.
   }
 
+  struct SignatureConfigConfig {
+    uint8 F; //  Maximum number of faulty/dishonest oracles the system can tolerate.
+    EnumerableSet.AddressSet signers;
+  }
+
   struct SignatureProof {
     bytes32[] rs; // R components of the signatures.
     bytes32[] ss; // S components of the signatures.
   }
 
-  SignatureConfigConfig internal s_ocrConfig;
-
-  EnumerableSet.AddressSet internal s_signers;
+  mapping(bytes32 configDigest => SignatureConfigConfig config) private s_signatureConfig;
```

Comment:
Do we have a TODO to transition to no config digest? Think that was a discussion recently cc @connorwstein